### PR TITLE
Migrate iam_role* modules and tests

### PR DESCRIPTION
**Pull request by abikouo**
SUMMARY

Migrate modules iam_role and iam_role_info from community.aws

ISSUE TYPE

    New Module Pull Request

COMPONENT NAME

iam_role
iam_role_info

### CODE REVIEW

File: changelogs/fragments/migrate_iam_role.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+major_changes:
+- iam_role - The module has been migrated from the ``community.aws`` collection. Playbooks
+  using the Fully Qualified Collection Name for this module should be updated to use
+  ``community.aws.iam_role``.
```
Comment (tremble) [inline changelogs/fragments/migrate_iam_role.yml:4]:
```suggestion
  ``amazon.aws.iam_role`` (https://github.com/ansible-collections/amazon.aws/pull/1760).
```

File: changelogs/fragments/migrate_iam_role.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- iam_role_info - The module has been migrated from the ``community.aws`` collection.
+  Playbooks using the Fully Qualified Collection Name for this module should be updated
+  to use ``community.aws.iam_role_info``.
```
Comment (tremble) [inline changelogs/fragments/migrate_iam_role.yml:7]:
```suggestion
  to use ``amazon.aws.iam_role_info`` (https://github.com/ansible-collections/amazon.aws/pull/1760).
```

**Review by tremble — CHANGES_REQUESTED** [review]
Migration, +1

Deprecations/Removals:
a) I don't like pulling this into the migration PR (too much changing at once)
b) Missing breaking changelog entry
c) Last time I spoke to @alinabuzachis 7.0.0 is hopefully landing early November (to meet the Ansible 9 release deadline), since the advertised date on the deprecation was 2023-12-01, we shouldn't include this in 7.0.0

Please clean-up the "new" commits (those that weren't from community.aws).  We need to remember to manually merge with commit without squashing too.

**Comment by abikouo** [comment]
> Migration, +1
> 
> Deprecations/Removals: a) I don't like pulling this into the migration PR (too much changing at once) b) Missing breaking changelog entry c) Last time I spoke to @alinabuzachis 7.0.0 is hopefully landing early November (to meet the Ansible 9 release deadline), since the advertised date on the deprecation was 2023-12-01, we shouldn't include this in 7.0.0
> 
> Please clean-up the "new" commits (those that weren't from community.aws). We need to remember to manually merge with commit without squashing too.

Pull request has been restricted to the code and tests migration, other (deprecation, unit tests) will follow later on another PR. The new commits have been squashed

**Review by tremble — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

File: plugins/modules/iam_role.py [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+
+from ansible_collections.amazon.aws.plugins.module_utils.arn import validate_aws_arn
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.policy import compare_policies
+from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import ansible_dict_to_boto3_tag_list
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import boto3_tag_list_to_ansible_dict
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import compare_aws_tags
+
+from ansible_collections.community.aws.plugins.module_utils.modules import AnsibleAWSModule
```
Comment (tremble) [inline plugins/modules/iam_role.py:236]:
```suggestion
from ansible_collections.amazon.aws.plugins.module_utils.modules import AnsibleAWSModule
```

**Review by tremble — CHANGES_REQUESTED** [review]

**Pull request merged**